src/User.js

```diff
@@ -1,39 +1,59 @@
 class User {
-  constructor(id, name) {
-    this.id = id;
-    this.name = name;
+  constructor(user) {
+    this.id = user.id;
+    this.name = user.name;
+    this.pantry = new Pantry(user.id, user.pantry);
     this.favoriteRecipes = [];
     this.recipesToCook = [];
   }
 
   storeRecipe(type, recipe) {
     this[type].push(recipe);
+    localStorage.setItem(type,
+      JSON.stringify(this[type]));
   }
 
   removeRecipe(type, recipe) {
     const index = this[type].findIndex(userRecipe => recipe === userRecipe);
     this[type].splice(index, 1);
+    localStorage.setItem(type,
+      JSON.stringify(this[type]));
   }
 
-  filterRecipe(type, tag, allRecipes) {
+  filterRecipes(type, tag) {
     switch (type) {
     case 'favoriteRecipes':
       return this.favoriteRecipes.filter(recipe => recipe.tags.includes(tag))
     case 'recipesToCook':
       return this.recipesToCook.filter(recipe => recipe.tags.includes(tag))
     default:
-      return allRecipes.filter(recipe => recipe.tags.includes(tag))
+      return recipeData.filter(recipe => recipe.tags.includes(tag))
     }
   }
 
-  searchRecipe(type, keyword, allRecipes) {
+  searchRecipes(type, keyword) {
     switch (type) {
     case 'favoriteRecipes':
       return this.favoriteRecipes.filter(recipe => recipe.ingredients.some(ingredient => ingredient.name.includes(keyword)))
     case 'recipesToCook':
       return this.recipesToCook.filter(recipe => recipe.ingredients.some(ingredient => ingredient.name.includes(keyword)))
     default:
-      return allRecipes.filter(recipe => recipe.ingredients.some(ingredient => ingredient.name.includes(keyword)))
+      return recipeData.filter(recipe => {
+        return (recipe.name.includes(keyword) || recipe.ingredients.some(ingredient => ingredient.name.includes(keyword)))
+      })
+    }
+  }
+
+  loadRecipes() {
+    //load both recipe objects from storage, objectify
+    const storageFavorite = JSON.parse(localStorage.getItem('favoriteRecipes'));
+    const storageCook = JSON.parse(localStorage.getItem('recipesToCook'));
+    //loop through, forEach instanciate as new recipes and push into user properties 
+    if (storageFavorite) {
+    storageFavorite.forEach(favorite => {this.favoriteRecipes.push(new Recipe(favorite))});
```

Comment:
You have unnecessary braces (`{}`) here. It should read:
`storageFavorite.forEach(favorite => this.favoriteRecipes.push(new Recipe(favorite)));`

Reply:
Removed, thanks!